### PR TITLE
Support arm64 builds

### PR DESCRIPTION
**Pull request by nbuchwitz**
This PR allows for optional arm64 builds of our piControl. Happy for your feedback :smile:

### CODE REVIEW

File: piIOComm.c [hunk 1]
```diff
@@ -413,7 +413,7 @@ int piIoComm_init(void)
 void piIoComm_finish(void)
 {
 	if (piIoComm_fd_m != NULL) {
-		pr_info_serial("filp_close %d\n", (int)piIoComm_fd_m);
+		pr_info_serial("filp_close %zd\n", (size_t)piIoComm_fd_m);
```
Comment (iluminat23) [inline piIOComm.c:416]: https://www.kernel.org/doc/html/latest/core-api/printk-formats.html#pointer-types

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: piIOComm.c [hunk 2]
```diff
@@ -207,7 +207,7 @@ int piIoComm_open_serial(void)
 	}
 	piIoComm_fd_m = fd;
 
-	pr_info_serial("filp_open %d\n", (int)piIoComm_fd_m);
+	pr_info_serial("filp_open %zd\n", (size_t)piIoComm_fd_m);
```
[the remark this answers is not in the document]
Comment (nbuchwitz) [reply piIOComm.c:210]: Other question @l1k and others: What is the purpose of this debug message? Is it probably better to just remove it if it is not used? The origin is a commit from MD 5 years ago and personally I'm not able to see the purpose of this kind of information...

File: piIOComm.c (hunk 1 above)
Comment (nbuchwitz) [reply piIOComm.c:416]: Other question @l1k and others: What is the purpose of this debug message? Is it probably better to just remove it if it is not used? The origin is a commit from MD 5 years ago and personally I'm not able to see the purpose of this kind of information...

File: piIOComm.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (linosanfilippo-kunbus) [reply piIOComm.c:210]: I think the address of the opened file is not very helpful. However the fact that the file has been opened successfully can help when tracking down issues. Maybe alter the message to something like pr_info_serial("file %s opened", REV_PI_TTY_DEVICE);

File: piIOComm.c (hunk 1 above)
Comment (linosanfilippo-kunbus) [reply piIOComm.c:416]: See my comment for the "file open" case.

File: piIOComm.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (nbuchwitz) [reply piIOComm.c]: Done in 3d29666395653a6d2bc54a6b38e4fb8125a42106

File: piIOComm.c (hunk 1 above)
Comment (nbuchwitz) [reply piIOComm.c]: Done in 3d29666395653a6d2bc54a6b38e4fb8125a42106

File: piFirmwareUpdate.c [hunk 3]
```diff
@@ -74,7 +74,7 @@ int FWU_update(tpiControlInst *priv, SDevice *pDev_p)
 
 	read = kernel_read(input, (char *)&header, sizeof(header), &input->f_pos);
 	if (read <= 0) {
-		pr_err("kernel_read returned %d: %x, %ld\n", read, (int)input, (long int)input->f_pos);
+		pr_err("kernel_read returned %d: %zx, %ld\n", read, (size_t)input, (long int)input->f_pos);
```
[the remark this answers is not in the document]
Comment (iluminat23) [reply piFirmwareUpdate.c:77]: `f_pos` is and `loff_t` which is a `__kernel__loff_t`
https://elixir.bootlin.com/linux/latest/source/include/linux/types.h#L46
https://elixir.bootlin.com/linux/latest/source/include/uapi/asm-generic/posix_types.h#L88
So `%lld` would be correct here.

**Review by linosanfilippo-kunbus — APPROVED** [review]
Looks good to me now

File: piFirmwareUpdate.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (nbuchwitz) [reply piFirmwareUpdate.c]: Thanks for looking this up!

**Review by iluminat23 — APPROVED** [review]

**Comment by l1k** [comment]
3d29666395653a6d2bc54a6b38e4fb8125a42106 will cause merge conflicts when merging `devel/revpi-5.10` into `master` because the code modified by 3d29666395653a6d2bc54a6b38e4fb8125a42106 was removed by b8d97c792425cca9040503669fa334008d532168.

I guess it goes to show that keeping a separate development branch for the serdev stuff is problematic. We probably shouldn't hold off much longer on merging the development branch into master.

**Comment by nbuchwitz** [comment]
Rebase: yes

Merge: Not until we figured out, where the increased cycle time is coming from. Otherwise this will prevent us from publishing updated kernel packages (dt changes or hotfixes like the io stop thingy)

**Pull request merged**